### ISSUE TITLE
Rebase

### ISSUE THREAD

**Issue by wrynvera** [Craigs-Crew/edgy-dark-ominous-game#29]
As you all know im working on the rebase.
For reference, I'm using the latest minetest_game as a base.
Progress is decent, so far, I have successfully separated all components of default into their respective locations.

In the next step, I'm eyeing to combine further minetest_game mods into the core mods.
flowers > flora
screwdriver > tools
sethome / game_commands / spawn > core (if the commands clash with craig's, ill reuse craig's ones)
wool > blocks
vessels > blocks

For future, once the furniture mod is done, signs, chests, vessels and bookshelves etc, should all be moved into the furniture mod, as per discussed in the furniture thread but for now, ill store them in blocks.

For compatibility purposes as well, instead of moving the various default functions into core, i have kept them in default for now.
I am however left with the lavacooling function, that I'm not sure we are including this time round, considering that was completely disabled on craigs

For stairs/fences/walls/xpanes, I'm awaiting #7 , once that is confirmed, it will go into shapes, a mega api mod for handling such registrations, for ease of mass registrations of such nodes than depending on multiple apis.

If all the above is agreeable, i should have the base in by tomr.

@lisacvuk @lumberjackgames

**Comment by wrynvera** [comment]
All the above I stated have been completed.
Except the lavacooling & stairs/fences/walls/xpanes, which I'm still awaiting further opinions on.

Awaiting further approval before I merge the first part of the base.

**Comment by lisacvuk** [comment]
I believe Craigs had two sethomes, one with the command, and one with the UI. Do we want to keep both? I feel that the rest is fine.

**Comment by wrynvera** [comment]
oh, yeah, i forgot to mention about that part, i took another look at craig_server commands and stuff, ill push back those for a while, i want to rewrite parts of it, will open another issue to discuss that when i get to it.
For now, those aren't essential anyways, so ill put it on hold.

**Comment by ChandlerBing1994** [comment]
Looks good, I believe it's ready to merge.

**Comment by wrynvera** [comment]
Merged. Keeping this issue open until you guys have the chance to fully review the structure accordingly.
For most parts, I separated nodes in various files accordingly to their themes.
The texture folder in blocks is separated into further folders according to the files, because I felt it was too messy with everything scattered.
Anything generating related goes into mapgen, this includes tree generation, grass and moss spread.

**Comment by wrynvera** [comment]
Also while going through the code, I do see the potential to improve a lot of them and remove redundancy, for those, ill submit pulls as appropriate.
It's mostly minor improvements so I don't see the need for starting an issue to discuss every single one, once one of your approves the respective pulls, ill merge accordingly.

**Comment by lumberjackgames** [comment]
Overall, looks good :+1:

>I believe Craigs had two sethomes, one with the command, and one with the UI. Do we want to keep both? I feel that the rest is fine.

I think having a gui for sethome is nice for mobile players but we can disable the command. I don't mind giving people a freebee personally for home/bottom of the mine, though. It's kind of nice when you are starting out and can't afford travelnets yet. As building is the ultimate goal (not pain) I think giving people a little help there would be nice, but we can settle that when we get to rewriting it.

**Comment by wrynvera** [comment]
Since there's no further comments, ill close this and mark it as completed.
If there's anything else, feel free to discuss with me in discord.
